datapipelines/src/main/kotlin/io/customer/datapipelines/migration/TrackingMigrationProcessor.kt

```diff
@@ -67,69 +56,86 @@ internal class TrackingMigrationProcessor(
     }
 
     override suspend fun processTask(task: MigrationTask): Result<Unit> = runCatching {
-        val trackEvent = when (task) {
-            is MigrationTask.IdentifyProfile -> IdentifyEvent(
-                userId = task.identifier,
-                traits = task.attributes.toJsonObject()
-            )
+        data class MigrationEventData(
+            val trackEvent: BaseEvent,
+            val enrichmentClosure: EnrichmentClosure? = null
+        )
```

Comment:
Why do we need to create the class every time processTask is called? 🤔 

We can make this class private if we don't want to expose to other modules, or use simple pair/map or similar class if we want to avoid adding new class just for migration.